pkl-executor/src/main/java/org/pkl/executor/spi/v1/ExecutorSpiOptions.java

```diff
@@ -83,6 +91,14 @@ public List<Path> getModulePath() {
     return modulePath;
   }
 
+  public List<Path> getCertificateFiles() {
+    return certificateFiles;
+  }
+
+  public List<URL> getCertificateUrls() {
+    return certificateUrls;
+  }
+
```

Comment:
I think I'd prefer to not change this.

If we add this, we'll need to introduce `v2`. And, for the executor use-case, you likely want Pkl to trust the same certs that the JVM already trusts anyways. It doesn't seem compelling enough to add a new knob here just yet.

I think let's have `ExecutorSpiImpl` use an http client that uses the default `SSLContext`. And in `EmbeddedExecutorTest`, we can use the good ol' `CertificateUtils.setupAllX509CertificatesGlobally` to get those tests to pass. Except move that class into `pkl-commons-test`.

Reply:
I don't mind not supporting certificates in Executor. But I really feel that `CertificateUtils.setupAllX509CertificatesGlobally` needs to go. It's a gross hack that's asking for trouble (might get used elsewhere, might interfere with other tests, code duplication with JdkHttpClient, etc.).

PS: My `package-server` PR also breaks `spi.v1`. Perhaps it's time for `spi.v2` anyways?

Reply:
By moving `CertificateUtils.setupAllX509CertificatesGlobally` into `pkl-commons-test`, we ensure that it's only used for testing purposes. And likely, the embedded executor test will be the only place that uses it, so we can even just inline its contents into that test itself.

> PS: My package-server PR also breaks spi.v1. Perhaps it's time for spi.v2 anyways?

Hm, I haven't reviewed that one yet. I'll need to take a look. Yeah if we do introduce spi.v2, might as well keep these options.

Reply:
> By moving CertificateUtils.setupAllX509CertificatesGlobally into pkl-commons-test, we ensure that it's only used for testing purposes.

I think it's only a matter of time until this will once again cause flaky tests. Mutating global state is best avoided. I'd be more than happy to implement `spi.v2` if that's what it takes.